hack/update-taps.sh

```diff
@@ -0,0 +1,253 @@
+# On 'release-x.y' branch the latest patch version for 'x.y.*' is returned
+# Similar to hack/library.sh get_latest_knative_yaml_source()
+function get_latest_release_version() {
+    local org_name="$1"
+    local repo_name="$2"
+    local major_minor=""
+    local version
+    version="$(git ls-remote --tags --ref https://github.com/"${org_name}"/"${repo_name}".git \
+      | grep "${major_minor}" \
+      | cut -d '-' -f2 \
+      | cut -d 'v' -f2 \
+      | sort -Vr \
+      | head -n 1)"
+    echo "${version}"
+}
+
+# Return the major version of a release.
+# For example, "v0.2.1" returns "0"
+# Parameters: $1 - release version label.
+function major_version() {
+  local release="${1//v/}"
+  local tokens=(${release//\./ })
+  echo "${tokens[0]}"
+}
+
+# Return the minor version of a release.
+# For example, "v0.2.1" returns "2"
+# Parameters: $1 - release version label.
+function minor_version() {
+  local tokens=(${1//\./ })
+  echo "${tokens[1]}"
+}
+
+# Return the release build number of a release.
+# For example, "v0.2.1" returns "1".
+# Parameters: $1 - release version label.
+function patch_version() {
+  local tokens=(${1//\./ })
+  echo "${tokens[2]}"
+}
+
+# Generate file based on the inlined HEREDOC template.
+function generate_tap_file() {
+  local out=$1
+  local name=${1%%.*}
+  name=${name%%@*}
+  local version=$2
+  local old_formula=${3:-""}
+  local darwin_amd64_checksum darwin_arm64_checksum linux_amd64_checksum linux_arm64_checksum
+  local darwin_amd64_binary darwin_arm64_binary linux_amd64_binary linux_arm64_binary
+
+  # Fetch checksum file from release and parse values
+  checksums=$(mktemp)
+  repo_url="https://github.com/knative-extensions/kn-plugin-${name}"
+  darwin_amd64_binary="kn-${name}-darwin-amd64"
+  darwin_arm64_binary="kn-${name}-darwin-arm64"
+  linux_amd64_binary="kn-${name}-linux-amd64"
+  linux_arm64_binary="kn-${name}-linux-arm64"
+  extra_args=""
+
+  if [[ "${name}" == "func" ]]; then
+    repo_url="https://github.com/knative/${name}"
+    darwin_amd64_binary="${name}_darwin_amd64"
+    darwin_arm64_binary="${name}_darwin_arm64"
+    linux_amd64_binary="${name}_linux_amd64"
+    linux_arm64_binary="${name}_linux_arm64"
+
+    extra_args='ln_s "kn-func", bin/"func"'
+  fi
+
+  curl -fsSL "${repo_url}/releases/download/knative-v${2}/checksums.txt" > "$checksums"
+  darwin_amd64_checksum=$(awk '$2=="'${darwin_amd64_binary}'"{print $1}' "$checksums")
+  darwin_arm64_checksum=$(awk '$2=="'${darwin_arm64_binary}'"{print $1}' "$checksums")
+  linux_amd64_checksum=$(awk '$2=="'${linux_amd64_binary}'"{print $1}' "$checksums")
+  linux_arm64_checksum=$(awk '$2=="'${linux_arm64_binary}'"{print $1}' "$checksums")
+  # Cleanup temp file
+  rm "${checksums}"
+
+  # Sanitize name to be class name
+  local className=${name^}
+  if [[ "${name}" == "source"* ]]; then
+    className=$(echo "${name}" | tr "-" " " | sed -e 's/\b./\U&/g' | tr -d " ")
+  fi
+
+  if [[ "${name}" == "event" ]]; then
+cat <<EOF > "$out"
```

Comment:
If possible to get rid of this special handling that would be great. I can't really judge whether this works (because I also don't fully understand why it was there in the first place), but rely on your judgment here.

Reply:
Sure the details are a bit blur. As I remember, those formula files require URLs and shas hardcoded. The special case of event plugin was introduced as an alternative approach to that. Resolving everything from version string only. The benefit is bumping just a version string. On the other hand, you download checksum together with binary that might lead to compromised server in the middle etc. But it ain't that strong security anyway. Trying to think about the few angles it aimed to solve. I'll redo the files to common format, though.